[SQL Logic Test] Add support for environment variables

Tishj opened the pull request:
This PR adds support to the unittest for using environment variables in sql

Previously `require-env` required 2 arguments:
`name` and `expected_value`

Now optionally you can leave out the `expected_value`, in which case we only verify that the environment variable is defined.

On top of that, we also now register these environment variables and allow tests to use them in SQL

## Example

```bash
➜  duckdb git:(master) ✗ export env_var_test=1                               
➜  duckdb git:(master) ✗ echo $env_var_test                            
```      

```sql
require-env env_var_test

query I
select ${env_var_test}
----
1
```

Note:
This does not yet work in a result list
```sql
query I
select 1;
----
${env_var_test}
```

Should probably also add this

Tishj committed:
add support for environment variables through expanding 'require-env'

Tishj commented:
Just realized I need to check for name collision with loops/foreach parameters

samansmink commented on test/sqlite/sqllogic_test_runner.cpp:
```diff
@@ -515,19 +521,35 @@ void SQLLogicTestRunner::ExecuteFile(string script) {
+			}
+
+			if (token.parameters.size() != 1 && token.parameters.size() != 2) {
 				parser.Fail("require-env requires 2 arguments: <env name> <env value>");
```
error message should probably explain both usages with 1 or 2 params

samansmink reviewed (commented):
added a small comment, also a test would indeed be nice. 

Maybe this is a good opportunity to switch all S3 test cases to use env variables from https://github.com/duckdb/duckdb/pull/5419 instead of the SET variables. Using the new require-env we could instead of checking for S3_TEST_SERVER_AVAILABLE, check for the presence of the AWS credentials in the env. This would also allow easy switching between s3 backends when testing.

Tishj committed:
improved usage message for 'require-env'

Tishj committed:
Merge branch 'master' into sqllogictest_env

Tishj committed:
adjusted existing tests to use require-env instead

Tishj committed:
add script to set s3 test_server variables, also added the variables to LinuxRelease.yml

Tishj commented:
I've adjusted all existing [s3] tests::
1. to use require-env to require `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_DEFAULT_REGION` environment variables
2. removed the initial SET statements for these variables
3. replaced the hardcoded later SETs (for resetting the variables) to use the ${env_var} functionality

Mytherin commented:
Thanks for the PR! LGTM. @samansmink any more thoughts?

samansmink commented on test/sql/copy/s3/metadata_cache.test:
```diff
@@ -18,7 +27,7 @@ statement ok
 
 statement ok
-SET s3_secret_access_key='minio_duckdb_user_password';SET s3_access_key_id='minio_duckdb_user';SET s3_region='eu-west-1'; SET s3_endpoint='duckdb-minio.com:9000'; SET s3_use_ssl=false;
+SET s3_endpoint='duckdb-minio.com:9000'; SET s3_use_ssl=false;
```
Could this s3 endpoint also be moved into an env variable? then we can use the env vars to easily switch between s3 environments for all tests

Tishj replied:
sure, domain + port together or separately?

samansmink replied:
I would say together is probably nicer

Tishj replied:
Now that we're looking at these variables, `SET s3_use_ssl=false;` also comes up a lot
maybe this should be an environment variable?

samansmink reviewed (commented):
Thanks for the changes, looks good! I added one small comment

Tishj committed:
also add 's3_endpoint' as env_var under DUCKDB_S3_ENDPOINT

Tishj committed:
replace more usage of the default values with their respective environment variables

Tishj committed:
also use require-env for the 's3_use_ssl' option

Mytherin commented:
Looks good from my side, @samansmink any more comments or is this ready to go?

samansmink commented:
@Mytherin lgtm!
